llvm/test/CodeGen/AIE/aie2/GlobalISel/duplicate-iterators.mir

```diff
@@ -0,0 +1,314 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
+#
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+
+# RUN: llc -mtriple aie2 -run-pass=aie-post-select-optimize %s -o - | FileCheck %s
+# RUN: llc -mtriple aie2 -start-before=aie-post-select-optimize -stop-after=register-coalescer \
+# RUN:      %s -o - | FileCheck %s -check-prefix=POST-COALESCER
+# RUN: llc -mtriple aie2 -start-after=aie-post-select-optimize -stop-after=register-coalescer \
+# RUN:      %s -o - | FileCheck %s -check-prefix=COALESCER
+
```

Comment:
Nit: maybe give some context. I think the `COALESCER` label is here to show what happens in the loop without duplicating the iterators.

Reply:
Done! I think it is a bit more clear now.